src/devices/elro_db286a.c

```diff
@@ -0,0 +1,104 @@
+/* Initial doorbell implementation for Elro DB286A devices
+ * 
+ * Note that each device seems to have two id patterns, which alternate
+ * for every other button press.
+ */
+ 
+#include "rtl_433.h"
+#include "pulse_demod.h"
+#include "data.h"
+#include "util.h"
+
+//33 pulses per data pattern
+#define	DB286A_PULSECOUNT		33
+//One leading zero pulse + 15*33 data pattern - 1 missing trailing zero pulse
+#define	DB286A_TOTALPULSES		DB286A_PULSECOUNT*15
+//Minimum data pattern repetitions (14 is maximum)
+#define	DB286A_MINPATTERN		5
+
+static int doorbell_db286a_callback(bitbuffer_t *bitbuffer) {
+	
+	char time_str[LOCAL_TIME_BUFLEN];
+	data_t *data;
+	bitrow_t *bb = bitbuffer->bb;
+	uint8_t *b = bb[0];
+	unsigned bits = bitbuffer->bits_per_row[0];
+
+	if (bits != DB286A_TOTALPULSES) {
+		return 0;
+	}
+	
+	//33 pulses + trailing null byte for C string
+	//Example pattern: 001101111111011000101010011011001
+	
```

Comment:
I tried to declare the processing variables only after making sure that the bitbuffer is going to be processed, but I see your point in readability - I will move it in a new commit.